src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -42,6 +46,27 @@ internal class JsonSchemaTransformer
             _xrefResolver = xrefResolver;
             _errors = errors;
             _monikerProvider = monikerProvider;
+
+            uidReferenceCountList = new ConcurrentBag<(SourceInfo<string>, string?, int?, int?)>();
+            xrefList = new ConcurrentBag<SourceInfo<string>>();
+        }
+
+        public void PostValidate()
+        {
+            foreach (var (uid, propertyPath, minReferenceCount, maxReferenceCount) in uidReferenceCountList)
+            {
+                var references = xrefList.Where(xref => xref.Value.Equals(uid.Value)).Select(xref => xref.Source);
```

Comment:
Add `ToArray` because each `Count()` below will re-evaluate the `Where` and `Select`.
```suggestion
                var references = xrefList.Where(xref => xref.Value.Equals(uid.Value)).Select(xref => xref.Source).ToArray();
```